### PR TITLE
Added optional outer clause to remove unnecessary parens in AND, OR clauses

### LINKED ISSUE #168: Too many brakets - parameter/customizable?
Hi!

Why does every AND operation add brakets to the query? I let ormlite generate a query and in many cases the statement becomes unfit for SQLite (on Android), because there are too many open brakets (if it is more than 8, it is game over).

The class ManyClause in the appendSql Method ALWAYS adds first an open braket and at the end a closing braket.

Would be great to have an option to change that behavior or a parameter.

Is that possible? Or is that in some way customizable?

### PR DESCRIPTION
**Pull request by j256**
Thanks to devjta.   Fixes #168.

### CODE REVIEW

File: src/main/javadoc/doc-files/changelog.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 	* CORE: More improvements to the TransactionManager connection and savepoint handling.
 	* CORE: Added a getInternalConnection() method to DatabaseConnection to access underlying objects.
 	* CORE: Added support for ORDER BY ... NULLS FIRST and LAST.  May not be supported by all database types.
+	* CORE: Added optimization to AND and OR queries to reduce the parens in the genrated query.  Thanks to devjta.
```
Comment (noordawod) [inline src/main/javadoc/doc-files/changelog.txt:5]: .. in the **generated** query

File: src/main/java/com/j256/ormlite/stmt/query/ManyClause.java [hunk 2]
```diff
@@ -51,22 +51,30 @@ public ManyClause(Clause[] others, Operation operation) {
...
-		sb.append('(');
-		first.appendSql(databaseType, tableName, sb, selectArgList);
+			List<ArgumentHolder> selectArgList, Clause outer) throws SQLException {
+		boolean closing;
+		if (outer instanceof ManyClause && ((ManyClause) outer).operation == operation) {
+			closing = false;
+		} else {
+			sb.append('(');
+			closing = true;
+		}
```
Comment (noordawod) [inline src/main/java/com/j256/ormlite/stmt/query/ManyClause.java:61]: Consider rewriting:

```java
boolean closing = !(outer instanceof ManyClause && ((ManyClause) outer).operation == operation);
if (closing) {
  sb.append('(');
}
```

**Review by noordawod — COMMENTED** [review]
LGTM. Code runs as is nicely. Left a couple of comments, though.

File: src/main/java/com/j256/ormlite/stmt/query/ManyClause.java (hunk 2 above)
Comment (j256) [reply src/main/java/com/j256/ormlite/stmt/query/ManyClause.java:61]: Ha!  I flip flopped over that particular line but I like the look of yours.  Hummm.

**Pull request merged**